keps/sig-api-machinery/2211-strip-managed-fields-server-side/README.md

```diff
@@ -0,0 +1,426 @@
+(such as kubectl or CLI tools written by users using client-go) to do the same thing.
+
+With this KEP, the API server could return objects without managedFields on demand, so that least client-side changes is needed
+to make the response less verbose. 
+
+### Goals
+
+* Strip managedFields on the server-side.
+
+### Non-Goals
+
+* Turn off managedFields. Server-side Apply relies on managedFields, we could not simply disable it.
+* Clear manageFields only on the client-side.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### Server-side changes
+
+* Add three content types to denote the client expects the API server to strip the managedFields.
+  * `application/json+unmanaged`
+  * `application/yaml+unmanaged`
+  * `application/vnd.kubernetes.protobuf+unmanaged`
+* Add a new serializer, say `UnmanagedSerializer`, which is able to strip the managedFields when encoding objects.
+* Make the API server to use `UnmanagedSerializer` when encounters the above three content type.
+
+### Client-side changes
+
+FIXME: we need to set the `Accept` header on the client-side and perform client-side stripping if the API server still
+returns objects with managedFields. I am not sure we should do this in `client-go` and `cli-runtime` or only in `kubectl get`.
+
+/cc @apelisse
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+TBD
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Server-side
+
+1. Add the three content types which represents the "unmanaged" format of objects to `k8s.io/apimachinery/pkg/runtime/types.go`:
+```go
+const (
+	ContentTypeJSON     string = "application/json"
+	ContentTypeYAML     string = "application/yaml"
+	ContentTypeProtobuf string = "application/vnd.kubernetes.protobuf"
+
+	ContentTypeUnmanagedJSON     string = "application/json+unmanaged"
+	ContentTypeUnmanagedYAML     string = "application/yaml+unmanaged"
+	ContentTypeUnmanagedProtobuf string = "application/vnd.kubernetes.protobuf+unmanaged"
```

Comment:
I don't think we should encourage a combinatorial explosion. We should represent this with a {json/yaml/proto, managed-fields:v1/omit} pair.

Reply:
@lavalamp Could you elaborate more on the representation please?